swarm/network/stream/syncer.go

```diff
@@ -188,16 +201,16 @@ func NewSwarmSyncerClient(_ *Peer, db *storage.DBAPI, ignoreExistingRequest bool
 // RegisterSwarmSyncerClient registers the client constructor function for
 // to handle incoming sync streams
 func RegisterSwarmSyncerClient(streamer *Registry, db *storage.DBAPI) {
-	streamer.RegisterClientFunc("SYNC", func(p *Peer, _ string, love bool) (Client, error) {
-		return NewSwarmSyncerClient(p, db, true)
+	streamer.RegisterClientFunc("SYNC", func(p *Peer, t string, live bool) (Client, error) {
+		return NewSwarmSyncerClient(p, db, true, NewStream("SYNC", t, live))
 	})
 }
 
 // NeedData
 func (s *SwarmSyncerClient) NeedData(key []byte) (wait func()) {
-	chunk, created := s.db.GetOrCreateRequest(key)
+	chunk, _ := s.db.GetOrCreateRequest(key)
 	// TODO: we may want to request from this peer anyway even if the request exists
-	if chunk.ReqC == nil || (s.ignoreExistingRequest && !created) {
+	if chunk.ReqC == nil { //|| (s.ignoreExistingRequest && !created) {
```

Comment:
can we already say if the commented part can definitively be removed now?

Reply:
We need recheck ignore existing request functionality, as it creates problems with the test. For now, we decided to remove the functionality. Will add some comments around it.